File: Progress-sheet/0x04-Sorting/maximum-average-subarray-i.cpp
using namespace std;
class Solution {
public:
    double findMaxAverage(vector<int>& nums, int k) {
        
        
        double sum = 0;
        for (int i = 0; i < k ; i++)
        {
             sum += nums[i];
        }
        int i = 0, j = k;
        double maxav = sum/k;

        while ( j < nums.size()){
            sum -= nums[i++];
            sum += nums[j++];

            maxav = max(maxav, sum/k);
        }

        return maxav;
        


    }
};

File: Python_track/maximum-average-subarray-i.py
class Solution:
    def findMaxAverage(self, nums: List[int], k: int) -> float:
        n = len(nums)
        curSum = sum(nums[:k])
        left = 0
        right = k
        maxSum = curSum
        # print(curSum)
        while(right < n):

            curSum += nums[right]
            curSum -= nums[left]
            left += 1
            right += 1

            maxSum = max(maxSum,curSum)
            # print(maxSum)
        return maxSum/k

